rama-http/src/layer/auth/require_authorization.rs

```diff
@@ -267,34 +268,82 @@ impl<ResBody> fmt::Debug for Basic<ResBody> {
     }
 }
 
-impl<S, B, ResBody> ValidateRequest<S, B> for AuthorizeContext<Basic<ResBody>>
+/// Trait for authorizing requests.
+pub trait Authorizer: Send + Sync + 'static {
```

Comment:
Can we also please make this sealed by doing the actual logic in a private trait. e.g. something like:

```
pub trait Authorizer: sealed::AuthSeal {}

impl ...

mod sealed {
      pub(super) trait AuthSeal: Send + Sync + 'static {
             ... the actual methods
      }

      impl ...
}
```

This way nobody else can call its logic or impl it.

Reply:
As this is a pretty internal solution that is not really made for others to build upon. for anything more advanced you probably want to make your own Rama layer.